### PR TITLE
Support extra flags for exec command

### PR DESCRIPTION
**Pull request by rossjones**
Previously was not supporting RuntimeSettings or SpecSettings and so was not feature complete with `docker run`. Not supports these flags although they may need to change in future v2 cli changes.

### CODE REVIEW

**Review by simonwo — APPROVED** [review]

File: cmd/cli/exec/exec.go [hunk 1]
```diff
@@ -210,9 +314,12 @@ func addInlineContent(ctx context.Context, codeLocation string, job *models.Job)
 
 	target := "/code"
 
-	finfo, _ := os.Stat(absPath)
-	if !finfo.IsDir() {
-		target = fmt.Sprintf("/code/%s", finfo.Name())
+	if finfo, err := os.Stat(absPath); err != nil {
+		return fmt.Errorf("file '%s' not found", codeLocation)
```
Comment (frrist) [inline cmd/cli/exec/exec.go:318]: might be worth include the specific error returned here (unless this is being passed directly to the user) stat can fail for other reasons, although usually due to file not being found of course.

File: cmd/cli/exec/exec.go [hunk 2]
```diff
@@ -210,9 +314,12 @@ func addInlineContent(ctx context.Context, codeLocation string, job *models.Job)
...
+	} else {
+		if !finfo.IsDir() {
+			target = fmt.Sprintf("/code/%s", finfo.Name())
```
Comment (frrist) [inline cmd/cli/exec/exec.go:321]: possible candidate for `filepath.Join`

File: cmd/util/parse/parse.go [hunk 3]
```diff
@@ -92,3 +93,15 @@ func JobOutputs(ctx context.Context, outputVolumes []string) ([]model.StorageSpe
 
 	return returnOutputVolumes, nil
 }
+
+func StringSliceToMap(slice []string) (map[string]string, error) {
+	result := make(map[string]string)
+	for _, item := range slice {
+		key, value, err := flags.SeparatorParser("=")(item)
+		if err != nil {
+			return nil, fmt.Errorf("expected 'key=value', received invalid format for key-value pair: %s", item)
```
Comment (frrist) [inline cmd/util/parse/parse.go:102]: is it also worth including the error in this message?

**Review by frrist — COMMENTED** [review]
left a couple opinions post merge sorry I didn't get to review this sooner.

**Pull request merged**